packages/auth-sveltekit/src/server.ts

```diff
@@ -0,0 +1,735 @@
+            { httpOnly: true, path: "/" }
+          );
+
+          return redirect(
+            303,
+            path.split("/").pop() === "signup"
+              ? pkceSession.getHostedUISignupUrl()
+              : pkceSession.getHostedUISigninUrl()
+          );
+        }
+
+        case "emailpassword/verify": {
+          if (!onEmailVerify) {
+            throw new Error(
+              `'onEmailVerify' auth route handler not configured`
+            );
+          }
+          const verificationToken = searchParams.get("verification_token");
+          const verifier = cookies.get(this.options.pkceVerifierCookieName);
+          if (!verificationToken) {
+            return onEmailVerify({
+              error: new Error("no verification_token in response"),
+            });
+          }
+          if (!verifier) {
+            return onEmailVerify({
+              error: new Error("no pkce verifier cookie found"),
+              verificationToken,
+            });
+          }
+          let tokenData: TokenData;
+          try {
+            tokenData = await (
+              await this.core
+            ).verifyEmailPasswordSignup(verificationToken, verifier);
+          } catch (err) {
+            return onEmailVerify({
+              error: err instanceof Error ? err : new Error(String(err)),
+              verificationToken,
+            });
+          }
+
+          cookies.set(this.options.authCookieName, tokenData.auth_token, {
+            httpOnly: true,
+            sameSite: "strict",
+            path: "/",
+          });
+
+          return onEmailVerify({
+            error: null,
+            tokenData,
+          });
+        }
+
+        case "signout": {
+          if (!onSignout) {
+            throw new Error(`'onSignout' auth route handler not configured`);
+          }
+          cookies.delete(this.options.authCookieName, { path: "/" });
+          return onSignout();
+        }
+
+        default:
+          throw new Error("Unknown auth route");
+      }
+    };
+  }
+
+  async emailPasswordSignUp(
+    req: Request,
+    cookies: Cookies,
+    data?: { email: string; password: string }
+  ): Promise<{ tokenData: TokenData }>;
+  async emailPasswordSignUp(
+    req: Request,
+    cookies: Cookies,
+    cb: (params: ParamsOrError<{ tokenData: TokenData }>) => CallbackReturn
+  ): Promise<CallbackReturn>;
+  async emailPasswordSignUp(
+    req: Request,
+    cookies: Cookies,
+    data: { email: string; password: string },
+    cb: (params: ParamsOrError<{ tokenData: TokenData }>) => CallbackReturn
+  ): Promise<CallbackReturn>;
+  async emailPasswordSignUp(
+    req: Request,
+    cookies: Cookies,
+    dataOrCb?:
+      | { email: string; password: string }
+      | ((params: ParamsOrError<{ tokenData: TokenData }>) => CallbackReturn),
+    cb?: (params: ParamsOrError<{ tokenData: TokenData }>) => CallbackReturn
+  ): Promise<
+    | {
+        tokenData: TokenData;
+      }
+    | CallbackReturn
+  > {
```

Comment:
I don't think we need to copy the all of this response wrapping code for the form actions from remix. As far as I can tell sveltekit is closer to nextjs, in that it lets you set cookies via the `cookies` object on the `RequestEvent` independently of what the user returns from the action function. So these form actions can be simplified to just taking the `RequestEvent` and any data they need and directly returning the `tokenData` (or whatever)/thowing an error.

Reply:
I agree about RequestEvent, but are we sure we don't wanna support callbacks?